### ISSUE TITLE
:path doesn't raise an error when the gem isn't present in the directory

### ISSUE THREAD

**Issue by indirect** [bundler/bundler#2511]
As reported by @faun, when using `:path` to point to a directory containing a .gem file (but not a gemspec or anything else that would make the gem work with the `:path` option), Bundler installed without complaint, but didn't provide the contents of the gem. Oops.

**Comment by agis-** [comment]
I couldn't reproduce this with 1.10.6:

```shell
agis:bundler-2511 $ cat Gemfile
source "https://rubygems.org"

gem "retriable", path: 'retriable'
agis:bundler-2511 $ mkdir retriable
agis:bundler-2511 $ mv ~/Downloads/retriable-2.1.0.gem retriable/
agis:bundler-2511 $ bundle --path vendor
Fetching gem metadata from https://rubygems.org/..
Fetching version metadata from https://rubygems.org/.
Could not find gem 'retriable' in source at `retriable`.
Source does not contain any versions of 'retriable'
```

I've also tried by cloning the repo in the specified path and removing the gemspec.

@indirect Is there anything that I'm missing or is this fixed?

**Comment by indirect** [comment]
Looks like we fixed this! I'm guessing it was part of 1.9. Thanks for testing it, @agis-.
